Examples in the docs: base PR for cleaning up notebook workflow (#303)
JuliaGaussianProcesses/KernelFunctions.jl: examples/README.md

# Example notebooks

The examples in this directory are stored in [Literate.jl](https://github.com/fredrikekre/Literate.jl) format.
To run them locally, simply `include("script.jl")` in the Julia REPL.

You can convert them to markdown and Jupyter notebook formats, respectively, by executing
```julia
julia> using Literate
julia> Literate.markdown("script.jl", "output_directory")
julia> Literate.notebook("script.jl", "output_directory")
```

## Add a new example

Create a new subdirectory in here, and put your code in a file called `script.jl` so that it will get picked up by the automatic docs build.

> You should also add a `Project.toml` file with all your script's dependencies (including KernelFunctions.jl).

Reviewer: ```suggestion

Every example uses a separate project environment. Therefore you should also create a new
project environment in the directory of the example, install all required package there, and
commit the `Project.toml` and a `Manifest.toml` file.
```

# Example notebooks

The examples in this directory are stored in [Literate.jl](https://github.com/fredrikekre/Literate.jl) format.

To run them locally, navigate to the directory with the example that you want to run and
start the Julia REPL and activate the project environment of the example:
```julia
julia> ] activate .
```
Alternatively, you can start Julia with `julia --project=.`. Then install all required
packages with
```julia
julia> ] instantiate
```
Afterwards simply run
```julia
julia> include("script.jl")
```

You can convert them to markdown and Jupyter notebook formats, respectively, by executing
```julia
julia> using Literate
julia> Literate.markdown("script.jl", "output_directory")
julia> Literate.notebook("script.jl", "output_directory")
```

> 

Author: ```suggestion
or run
`julia --project=. ../../docs/literate.jl script.jl output_directory`.

```

## Add a new example

Create a new subdirectory in here, and put your code in a file called `script.jl` so that it will get picked up by the automatic docs build.

Every example uses a separate project environment. Therefore you should also create a new

> project environment in the directory of the example, install all required package there (including KernelFunctions.jl), and

Author: ```suggestion
project environment in the directory of the example that contains all packages required by your script. Note that you must also include both the `KernelFunctions` and the `Literate` packages. Make sure to both
```

# Example notebooks

The examples in this directory are stored in [Literate.jl](https://github.com/fredrikekre/Literate.jl) format.

To run them locally, navigate to the directory with the example that you want to run and
start the Julia REPL and activate the project environment of the example:
```julia
julia> ] activate .
```
Alternatively, you can start Julia with `julia --project=.`. Then install all required
packages with
```julia
julia> ] instantiate
```
Afterwards simply run
```julia
julia> include("script.jl")
```

You can convert them to markdown and Jupyter notebook formats, respectively, by executing
```julia
julia> using Literate
julia> Literate.markdown("script.jl", "output_directory")
julia> Literate.notebook("script.jl", "output_directory")
```
or run
`julia --project=. ../../docs/literate.jl script.jl output_directory`.

## Add a new example

Create a new subdirectory in here, and put your code in a file called `script.jl` so that it will get picked up by the automatic docs build.

Every example uses a separate project environment. Therefore you should also create a new

> project environment in the directory of the example that contains all packages required by your script. Note that you must also include both the `KernelFunctions` and the `Literate` packages. Make sure to both
> commit the `Project.toml` and a `Manifest.toml` file.

Author: ``````suggestion
project environment in the directory of the example that contains all packages required by your script.
Note that the dependencies of your example must include the `Literate` package.
From a Julia REPL started in your example script's directory, you can run
```julia
julia> ] activate .
julia> ] add Literate
julia> ] add KernelFunctions
julia> # add any other example-specific dependencies
```
to generate the project files.
Make sure to commit both the `Project.toml` and the `Manifest.toml` file when you want to contribute your example in a pull request.
``````

# Example notebooks

The examples in this directory are stored in [Literate.jl](https://github.com/fredrikekre/Literate.jl) format.

To run them locally, navigate to the directory with the example that you want to run and
start the Julia REPL and activate the project environment of the example:
```julia
julia> ] activate .
```
Alternatively, you can start Julia with `julia --project=.`. Then install all required
packages with
```julia
julia> ] instantiate
```
Afterwards simply run
```julia
julia> include("script.jl")
```

You can convert them to markdown and Jupyter notebook formats, respectively, by executing
```julia
julia> using Literate
julia> Literate.markdown("script.jl", "output_directory")
julia> Literate.notebook("script.jl", "output_directory")
```
or run

> `julia --project=. ../../docs/literate.jl script.jl output_directory`.

Reviewer: These two approaches are not equivalent though, we use different keyword arguments. Maybe one should just mention `literate.jl` and not the manual conversion (IMO the recommended workflow for writing examples would not involve running Literate at all, so it seems fine to run it in a separate process if you actually want to convert the example).

It is also not necessary to use `--project=.`, the environment is activated by `literate.jl` automatically. However, I think the script does not work if you call it in this way since it extracts the name of the example from the path of `script.jl` which is not given. So it might be better to recommend to call
````suggestion
```shell
julia docs/literate.jl examples/myexample/script.jl output_directory
```
````

# Example notebooks

The examples in this directory are stored in [Literate.jl](https://github.com/fredrikekre/Literate.jl) format.

To run them locally, navigate to the directory with the example that you want to run and
start the Julia REPL and activate the project environment of the example:
```julia
julia> ] activate .
```
Alternatively, you can start Julia with `julia --project=.`. Then install all required
packages with
```julia
julia> ] instantiate
```
Afterwards simply run
```julia
julia> include("script.jl")
```

You can convert them to markdown and Jupyter notebook formats, respectively, by executing
```julia
julia> using Literate
julia> Literate.markdown("script.jl", "output_directory")
julia> Literate.notebook("script.jl", "output_directory")
```

> or run

Author: I've rewritten it a bit

# Example notebooks

The examples in this directory are stored in [Literate.jl](https://github.com/fredrikekre/Literate.jl) format.

To run them locally, navigate to the directory with the example that you want to run and
start the Julia REPL and activate the project environment of the example:
```julia
julia> ] activate .
```
Alternatively, you can start Julia with `julia --project=.`. Then install all required
packages with
```julia
julia> ] instantiate
```
Afterwards simply run
```julia
julia> include("script.jl")
```

> In particular when editing an example, it can be convenient to (re-)run only some parts of
> an example. Many editors with Julia support such as VSCode, Juno, and Emacs support the evaluation of individual lines or code chunks.
> You can convert them to markdown and Jupyter notebook formats, respectively, by executing
> ```julia
> julia> using Literate
> julia> Literate.markdown("script.jl", "output_directory")
> julia> Literate.notebook("script.jl", "output_directory")
> ```
> or run
> ```shell
> julia docs/literate.jl examples/myexample/script.jl output_directory
> ```

Author: ``````suggestion
In particular when editing an example, it can be convenient to (re-)run only some parts of
an example.
Many editors with Julia support such as VSCode, Juno, and Emacs support the evaluation of individual lines or code chunks.

You can convert a notebook to markdown and Jupyter notebook formats, respectively, by executing
```julia
julia> using Literate
julia> Literate.markdown("script.jl", "output_directory")
julia> Literate.notebook("script.jl", "output_directory")
```
(see the [Literate.jl docs](https://fredrikekre.github.io/Literate.jl/v2/) for additional options) or run
```shell
julia docs/literate.jl examples/myexample/script.jl output_directory
```
which also executes the code and generates embedded plots etc. in the same way as in building the KernelFunctions documentation.
``````
